api/client/commands.go

```diff
@@ -222,6 +251,9 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	if *pull {
 		v.Set("pull", "1")
 	}
+
+	v.Set("f", *dockerfileName)
```

Comment:
I mean that "f" which is the API parameter.  A third-party API client
wanting to use this feature would have to specify a query parameter of
"f=..." to use this feature, which isn't very intuitive for an API IMO (but
maybe I'm alone in this?  we have an absolute plethora of similarly
badly-named API parameters already).